File: DotnetAsm.Roslyn/Models/RoslynDiagnostic.cs
using Microsoft.CodeAnalysis;

namespace DotnetAsm.Roslyn.Models;

public class RoslynDiagnostic
{
    public required string Id { get; init; }
    public required string Message { get; init; }
    public required int OffsetFrom { get; init; }
    public required int OffsetTo { get; init; }
    public required DiagnosticSeverity Severity { get; init; }
}

File: DotnetAsm.Frontend/src/models/Roslyn/Diagnostics/RoslynDiagnostic.ts
import RoslynDiagnosticSeverity from "./RoslynDiagnosticSeverity";

type RoslynDiagnostic = {
  id: string;
  message: string;
  offsetFrom: number;
  offsetTo: number;
  severity: RoslynDiagnosticSeverity;
};

export default RoslynDiagnostic;
